src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -43,12 +50,14 @@ public sealed class PeerAddressManagerBehaviour : NetworkPeerBehavior
         /// </summary>
         private bool sentAddress;
 
-        public PeerAddressManagerBehaviour(IDateTimeProvider dateTimeProvider, IPeerAddressManager peerAddressManager)
+        public PeerAddressManagerBehaviour(IDateTimeProvider dateTimeProvider, IPeerAddressManager peerAddressManager, ILoggerFactory loggerFactory)
         {
             Guard.NotNull(dateTimeProvider, nameof(dateTimeProvider));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
 
             this.dateTimeProvider = dateTimeProvider;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName, $"[{this.GetHashCode():x}] ");
+            this.loggerFactory = loggerFactory;
```

Comment:
Do we need to save logger factory here or it's needed just to create logger once per class lifetime?

Reply:
It's used for Clone() down the bottom unfortunately

Reply:
AH, yes, indeed